backend/server.js

```diff
@@ -32,35 +32,49 @@ const createWorker = (path: string): {
   rpcSbp: (...args: any) => Promise<any>,
   terminate: () => Promise<number>
 } => {
-  const worker = new Worker(path)
-  const ready = new Promise((resolve, reject) => {
-    worker.on('message', (msg) => {
-      if (msg === 'ready') resolve()
-    })
-    worker.on('error', reject)
-  })
+  let worker
+  let ready: Promise<void>
 
-  const rpcSbp_ = (...args: any) => {
+  const launchWorker = () => {
+    worker = new Worker(path)
     return new Promise((resolve, reject) => {
+      worker.on('message', (msg) => {
+        if (msg === 'ready') {
+          worker.off('error', reject)
+          worker.on('error', (e) => {
+            console.error(e, 'Error on worker', path)
+            ready = launchWorker()
+            ready.catch(e => {
+              console.error(e, 'Error on worker relaunch', path)
+              process.exit(1)
+            })
```

Comment:
So this allows at most 1 worker relaunch per runtime instance?

Does that make sense to do? Can you add a comment here explaining rationale?

Reply:
No, this allows for multiple relaunches, but it terminates if relaunching immediately fails.